### DISCUSSION TITLE
黑暗模式和自适应样式

### DISCUSSION THREAD

**Discussion by Gk0Wk** [tiddly-gittly/TidGi-Desktop#124] [Ideas]
`src/services/theme/defaultTheme.ts`里面加一下：

```ts
foreground: {
  default: XXX,
},
```

然后添加一下`color`和`fill`样式描述。

---

进一步的，能否直接获得容器内TW的运行环境呀？如果可以的话，可以直接使用

```js
$tw.utils.parseFields($tw.wiki.getTiddlerText($tw.wiki.getTiddlerText('$:/palette')))
```

来获取tw内当前使用的配色方案，就可以让TG的配色方案随着TW的配色而改变，应该是一个很不错的功能。

**Comment by linonetwo** [comment]
1. 就是要改的地方比较多，得在各处引用这个 foreground 变量。
2. tw内当前使用的配色方案得在 wiki 启动后才能获取到，不过应该也可以，切换到哪个 wiki 就用哪个的配色。

目前是可以在 https://github.com/tiddly-gittly/TiddlyGit-Desktop/blob/master/src/preload/wikiOperation.ts 在 wiki 里执行 JS ，就是不知道能不能 await 到返回的结果，如果可以的话，那就在每次切换 wiki 的时候调一下，然后更新全局 theme 就好。

**Reply by linonetwo** [reply]
这个我估计下周可以搞搞，这周我在搞 menubar 的功能
